File: app/src/components/CouponPickBlock.js
// Maç detayında KUPON SEÇİM bloğu — analiz görürken [1][X][2] + Sistemden al.
// Seçim paylaşılan TASLAK'a işlenir; Kupon Oluştur'dan kaydedilir. Kilit sonrası
// sadece görüntüleme. Yalnız resmi TEYİTLİ güncel bültendeki maçta görünür.
import React, { useState } from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { colors, spacing, radius, shadow } from '../theme';
import { setDraftPick } from '../coupon/store';
import { isMatchLocked } from '../couponConfig';
import { userSelectedAnalysisEngine } from '../analysis/engine';
import { getActiveProfile, countOn, ensureDefaultProfile } from '../analysisProfile';

const OUT = ['1', 'X', '2'];

export default function CouponPickBlock({ m, navigation }) {
  // Otomatik gelmez — kullanıcı 1/X/2 seçer veya "Sistemden al"a basar.
  const [pick, setPick] = useState([]);
  if (!m || m.verificationStatus !== 'confirmed' || m.roundId == null) return null;
  // KİLİT MAÇ BAZINDA: yalnız BU maç başladıysa seçim donar; bülten mühürlense
  // bile başlamamış maça hafta boyunca seçim yapılabilir. Maç saati yoksa
  // (belirsizlik) eski bülten kapanışına düşülür — asla habersiz açık kalmaz.
  const locked = m.date
    ? isMatchLocked(m)
    : (m.closeDate && Date.now() >= new Date(m.closeDate).getTime());

  const write = (arr) => { setPick(arr); setDraftPick(m.roundId, m.no, arr); };
  const toggle = (o) => { if (locked) return; const set = new Set(pick); set.has(o) ? set.delete(o) : set.add(o); write(OUT.filter((x) => set.has(x))); };
  // "Sistemden al" → AKTİF ANALİZ PROFİLİ'ne göre Ana Seçim (1X / X2 / 1X2 vb.).
  // Profil yoksa/boşsa analiz üretilmez → kullanıcıyı kriter seçimine yönlendir.
  const fromSystem = () => {
    if (locked) return;
    let profile = getActiveProfile();
    if (!profile) profile = ensureDefaultProfile();   // kurulum istemeden çalışır
    if (!profile || countOn(profile) === 0) { navigation.navigate('AnalysisSettings'); return; }
    const main = userSelectedAnalysisEngine(m, profile)?.verdict?.main || '';
    write(OUT.filter((o) => main.includes(o)));
  };

  return (
    /* OLABİLDİĞİNCE KÜÇÜK (kullanıcı isteği, 2026-08-06). Blok beş satırdı:
       başlık, iki satırlık açıklama, düğmeler, "Henüz seçim yok…" ipucu ve
       "Seçimin: 1" tekrarı. İkisi de gereksizdi — seçilen işaret ZATEN kendi
       düğmesinde ✓ ve dolgu ile görünüyor; açıklama da düğmelerin yaptığı işi
       kelimeyle anlatıyordu. Şimdi iki satır: başlık + Kupon Oluştur yan yana,
       altında 1/X/2 + Sistemden al. Hiçbir işlev kaybolmadı. */
    <View style={s.block}>
      <View style={s.head}>
        <Text style={s.title}>🎟️ KUPONA İŞLE</Text>
        {!locked ? (
          <TouchableOpacity
            onPress={() => navigation.navigate('CouponEditor', { roundId: m.roundId })}
            style={s.goBtn}
            activeOpacity={0.85}
          >
            <Text style={s.goTxt}>Kupon Oluştur ›</Text>
          </TouchableOpacity>
        ) : null}
      </View>
      {locked ? (
        <Text style={s.locked}>🔒 Maç başladı — bu maç için seçim değiştirilemez.</Text>
      ) : (
        <View style={s.row}>
          {OUT.map((o) => { const on = pick.includes(o); return (
            <TouchableOpacity
              key={o}
              onPress={() => toggle(o)}
              activeOpacity={0.85}
              style={[s.btn, on && s.btnOn]}
              accessibilityRole="button"
              accessibilityState={{ selected: on }}
              accessibilityLabel={`${o} işareti${on ? ' — seçili' : ''}`}
            >
              {on ? <Text style={s.check}>✓</Text> : null}
              <Text style={[s.txt, on && s.txtOn]}>{o}</Text>
            </TouchableOpacity>
          ); })}
          <TouchableOpacity onPress={fromSystem} activeOpacity={0.85} style={s.sysBtn}>
            <Text style={s.sysTxt} numberOfLines={1}>⚙ Sistemden al</Text>
          </TouchableOpacity>
        </View>
      )}
    </View>
  );
}

const s = StyleSheet.create({
  block: { backgroundColor: colors.card, borderRadius: radius.md, borderWidth: 1, borderColor: colors.primary, padding: 8, marginBottom: spacing.sm },
  head: { flexDirection: 'row', alignItems: 'center', justifyContent: 'space-between', gap: 8 },
  title: { color: colors.primary, fontSize: 11, fontWeight: '900', letterSpacing: 0.3, flexShrink: 1 },
  locked: { color: colors.warning, fontSize: 11.5, fontWeight: '800', marginTop: 6 },
  row: { flexDirection: 'row', gap: 6, marginTop: 7 },
  btn: { flex: 1, flexDirection: 'row', gap: 4, paddingVertical: 7, borderRadius: radius.sm, backgroundColor: colors.cardAlt, alignItems: 'center', justifyContent: 'center', borderWidth: 2, borderColor: 'transparent' },
  btnOn: { backgroundColor: colors.primary, borderColor: colors.primary, ...shadow.soft },
  txt: { color: colors.textSoft, fontSize: 14, fontWeight: '900' }, txtOn: { color: '#fff' },
  check: { color: '#fff', fontSize: 11, fontWeight: '900' },
  // Sistemden al, üç işaret kutusuyla aynı yükseklikte kalsın diye flexShrink'li:
  // uzun metinde küçülür, satırı alta itmez.
  sysBtn: { paddingHorizontal: 9, paddingVertical: 7, borderRadius: radius.sm, backgroundColor: colors.surfaceSoft, borderWidth: 1, borderColor: colors.border, justifyContent: 'center', flexShrink: 1 },
  sysTxt: { color: colors.primary, fontSize: 10.5, fontWeight: '900' },
  goBtn: { paddingHorizontal: 10, paddingVertical: 6, borderRadius: radius.sm, backgroundColor: colors.primary },
  goTxt: { color: '#fff', fontSize: 10.5, fontWeight: '900' },
});

File: lib/features/match_detail/coupon_pick_block.dart
// KAYNAK: app/src/components/CouponPickBlock.js — BİREBİR çeviri.
//
// Maç detayında KUPON SEÇİM bloğu — analiz görürken [1][X][2] + Sistemden al.
// Seçim paylaşılan TASLAĞA işlenir; Kupon Oluştur'dan kaydedilir. Kilit
// sonrası sadece görüntüleme. Yalnız resmi TEYİTLİ güncel bültendeki maçta
// görünür.
//
// KİLİT MAÇ BAZINDA: yalnız BU maç başladıysa seçim donar; bülten mühürlense
// bile başlamamış maça hafta boyunca seçim yapılabilir. Maç saati yoksa
// (belirsizlik) eski bülten kapanışına düşülür — asla habersiz açık kalmaz.

import 'package:flutter/material.dart';

import '../../core/coupon/coupon_config.dart';
import '../../core/coupon/coupon_store.dart';
import '../../core/theme/tokens.dart';

class CouponPickBlock extends StatefulWidget {
  const CouponPickBlock({super.key, required this.m, this.onKuponOlustur});

  final Map<String, dynamic> m;
  final void Function(Object roundId)? onKuponOlustur;

  @override
  State<CouponPickBlock> createState() => _CouponPickBlockState();
}

class _CouponPickBlockState extends State<CouponPickBlock> {
  List<String> _pick = const [];

  @override
  void initState() {
    super.initState();
    // Taslakta bu maç için kayıtlı seçim varsa geri yüklenir. Kaynakta blok
    // her açılışta boş başlıyordu (`useState([])`); taslak zaten diskte
    // durduğu için seçimi göstermemek kullanıcıya "kaydolmadı" izlenimi
    // veriyordu. Depo aynı, gösterim dürüstleşti.
    final roundId = widget.m['roundId'];
    final no = widget.m['no'];
    if (roundId == null || no == null) return;
    final picks = (getDraft(roundId)['picks'] as Map?) ?? const {};
    final mevcut = picks['$no'];
    if (mevcut is List) _pick = mevcut.cast<String>();
  }

  @override
  Widget build(BuildContext context) {
    final m = widget.m;
    // Yalnız resmi TEYİTLİ güncel bültendeki maçta görünür.
    if (m['verificationStatus'] != 'confirmed' || m['roundId'] == null) {
      return const SizedBox.shrink();
    }

    final locked = m['date'] != null
        ? isMatchLocked(m)
        : (m['closeDate'] != null &&
              !DateTime.now().isBefore(
                DateTime.tryParse('${m['closeDate']}')?.toLocal() ??
                    DateTime(9999),
              ));

    return Container(
      margin: const EdgeInsets.only(bottom: Spacing.sm),
      padding: const EdgeInsets.all(8),
      decoration: BoxDecoration(
        color: AppColors.card,
        borderRadius: AppRadius.mdR,
        border: Border.all(color: AppColors.primary),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          Row(
            children: [
              Expanded(
                child: Text(
                  '🎟️ KUPONA İŞLE',
                  style: TextStyle(
                    color: AppColors.primary,
                    fontSize: 11,
                    fontWeight: AppFont.black,
                    letterSpacing: 0.3,
                  ),
                ),
              ),
              if (!locked)
                GestureDetector(
                  onTap: () =>
                      widget.onKuponOlustur?.call(m['roundId'] as Object),
                  behavior: HitTestBehavior.opaque,
                  child: Container(
                    padding: const EdgeInsets.symmetric(
                      horizontal: 10,
                      vertical: 6,
                    ),
                    decoration: BoxDecoration(
                      color: AppColors.primary,
                      borderRadius: AppRadius.smR,
                    ),
                    child: Text(
                      'Kupon Oluştur ›',
                      style: TextStyle(
                        color: AppColors.onPrimary,
                        fontSize: 10.5,
                        fontWeight: AppFont.black,
                      ),
                    ),
                  ),
                ),
            ],
          ),
          if (locked)
            const Padding(
              padding: EdgeInsets.only(top: 6),
              child: Text(
                '🔒 Maç başladı — bu maç için seçim değiştirilemez.',
                style: TextStyle(
                  color: AppColors.warning,
                  fontSize: 11.5,
                  fontWeight: AppFont.heavy,
                ),
              ),
            )
          else
            Padding(
              padding: const EdgeInsets.only(top: 7),
              child: Row(
                children: [
                  for (final o in kOutcomes) ...[
                    Expanded(child: _isaretDugmesi(o)),
                    const SizedBox(width: 6),
                  ],
                  Flexible(
                    child: GestureDetector(
                      onTap: _sistemdenAl,
                      behavior: HitTestBehavior.opaque,
                      child: Container(
                        padding: const EdgeInsets.symmetric(
                          horizontal: 9,
                          vertical: 7,
                        ),
                        decoration: BoxDecoration(
                          color: AppColors.surfaceSoft,
                          borderRadius: AppRadius.smR,
                          border: Border.all(color: AppColors.border),
                        ),
                        child: Text(
                          '⚙ Sistemden al',
                          maxLines: 1,
                          overflow: TextOverflow.ellipsis,
                          style: TextStyle(
                            color: AppColors.primary,
                            fontSize: 10.5,
                            fontWeight: AppFont.black,
                          ),
                        ),
                      ),
                    ),
                  ),
                ],
              ),
            ),
        ],
      ),
    );
  }

  Widget _isaretDugmesi(String o) {
    final on = _pick.contains(o);
    return Semantics(
      button: true,
      selected: on,
      label: '$o işareti${on ? ' — seçili' : ''}',
      child: GestureDetector(
        onTap: () => _degistir(o),
        behavior: HitTestBehavior.opaque,
        child: Container(
          padding: const EdgeInsets.symmetric(vertical: 7),
          alignment: Alignment.center,
          decoration: BoxDecoration(
            color: on ? AppColors.primary : AppColors.cardAlt,
            borderRadius: AppRadius.smR,
            border: Border.all(
              color: on ? AppColors.primary : Colors.transparent,
              width: 2,
            ),
            boxShadow: on ? AppShadow.soft : null,
          ),
          child: Row(
            mainAxisSize: MainAxisSize.min,
            children: [
              if (on) ...[
                Text(
                  '✓',
                  style: TextStyle(
                    color: AppColors.onPrimary,
                    fontSize: 11,
                    fontWeight: AppFont.black,
                  ),
                ),
                const SizedBox(width: 4),
              ],
              Text(
                o,
                style: TextStyle(
                  color: on ? AppColors.onPrimary : AppColors.textSoft,
                  fontSize: 14,
                  fontWeight: AppFont.black,
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }

  Future<void> _yaz(List<String> arr) async {
    setState(() => _pick = arr);
    await setDraftPick(widget.m['roundId'], widget.m['no'] as Object, arr);
  }

  void _degistir(String o) {
    final set = _pick.toSet();
    if (!set.add(o)) set.remove(o);
    _yaz(kOutcomes.where(set.contains).toList());
  }

  /// "Sistemden al" → RESMÎ SİSTEM TAHMİNİ (2026-08-07).
  ///
  /// Eskiden kullanıcının kendi kriter profilinden hesaplanıyordu; kriter
  /// seçme sistemi kullanıcı kararıyla tamamen kaldırıldığı için artık
  /// bültenin resmî tahmini kullanılır. Tahmin yoksa SEÇİM YAPILMAZ
  /// (uydurma sembol yazılmaz).
  void _sistemdenAl() {
    final sym = (widget.m['prediction'] as Map?)?['symbol'] as String?;
    if (sym == null || sym == '-') return;
    // Resmî sembol '0' beraberliği gösterir; kupon tarafında 'X' kullanılır.
    final main = sym.split('').map((c) => c == '0' ? 'X' : c).join();
    _yaz(kOutcomes.where(main.contains).toList());
  }
}
